en/ros/ros2.md

```diff
@@ -1,27 +1,28 @@
 # ROS 2
 
 [ROS 2](https://index.ros.org/doc/ros2/) is the newest version of [ROS](http://www.ros.org/) (Robot Operating System), a general purpose robotics library that can be used with the PX4 Autopilot to create powerful drone applications.
-It captures most of the learnings and recently added features of [ROS (1)](../ros/ros1.md), improving a number of flaws of the earlier version.
+It captures most of the learnings and features of [ROS (1)](../ros/ros1.md), improving a number of flaws of the earlier version.
 
 :::warning Tip
 The PX4 development team highly recommend that you use/migrate to this version of ROS!
 :::
 
-The translation layer between ROS 2 and PX4 is software known as the [PX4-ROS 2 bridge](../ros/ros2_comm.md).
-This provides a bridge between PX4 UORB messages and ROS 2 messages and types, effectively allowing direct access to PX4 from ROS 2 workflows and nodes.
-The bridge uses UORB message definitions and correspondent IDL types to generate code to serialise and deserialise the messages heading in and out of PX4.
+Communication between ROS 2 and PX4 uses middleware that implements the [XRCE-DDS protocol](../middleware/xrce_dds.md).
+This middleware exposes PX4 [uORB messages](../msg_docs/README.md) as ROS 2 messages and types, effectively allowing direct access to PX4 from ROS 2 workflows and nodes.
+The middleware uses UORB message definitions to generate code to serialise and deserialise the messages heading in and out of PX4.
+These same message definitions are used in ROS2 applications to allow the messages to be interpretted.
 
 The main topics in this section are:
-- [ROS 2 User Guide](../ros/ros2_comm.md): an overview how to use ROS 2 with PX4 (covering the PX4-ROS2 bridge, installation/setup, and how to build ROS 2 applications for PX4).
-- [ROS 2 microRTPS Offboard Control Example](../ros/ros2_offboard_control.md)
+- [ROS 2 User Guide](../ros/ros2_comm.md): A PX4-centric overview of ROS2, covering installation, setup, and how to build ROS 2 applications that communicate with PX4.
+- [ROS 2 Offboard Control Example](../ros/ros2_offboard_control.md)
 
 :::note
 ROS 2 is officially supported only on Linux platforms.
 Ubuntu 20.04 LTS is the official supported distribution.
 :::
 
 :::note
-To use the [PX4-ROS 2 bridge](../ros/ros2_comm.md) effectively you must (at time of writing) have a reasonable understanding of the PX4 internal architecture and conventions.
+To use the [ROS2](../ros/ros2_comm.md) effectively you must (at time of writing) have a reasonable understanding of the PX4 internal architecture and conventions.
```

Comment:
@beniaminopozzan This statement (and the ones below it, not all of which are visible here) states that you need know a bit about PX4 conventions to use ROS2 properly. This was added when ROS 2 first came out because at that time MAVROS did all the conversions for conventions - e.g. FRD to FRU and visa versa between ROS2 and PX4.

Is this still true/an issue or are conventions now taken care of for ROS 2 users?

Assuming this is a problem, do we know what the issues are? If we're not going to provide translation layers we should document the different conventions.

Reply:
The statement still holds. In details, most of the uorb messages assume implicitly that PX4 conventions are used (after all, they where not supposed to be exposed to external components such as ROS nodes), some of them, which are quite relevant are https://github.com/PX4/PX4-Autopilot/blob/main/msg/TrajectorySetpoint.msg, https://github.com/PX4/PX4-Autopilot/blob/main/msg/VehicleLocalPositionSetpoint.msg.
Therefore it is necessary that the user converts properly the frames before using them (like taking off sending negative z-position setpoints)

On the other hand, at least https://github.com/PX4/PX4-Autopilot/blob/main/msg/VehicleOdometry.msg (and all the topics that uses this message) allows for different frames. Sadly, ROS2 conventions are still not supported, but there is a draft PR to fix that: https://github.com/PX4/PX4-Autopilot/pull/20737. Nevertheless, this will allow to send data in ROS2 convention to PX4, but PX4 will always reply using PX4 convention.

> Assuming this is a problem, do we know what the issues are?

I wouldn't call this a problem, it's more a _fundamental difference_ that has to be clarified :)
On the implementation POV, it is quite straightforward once you know which rotation to apply. Moreover, lets assume you have a ROS node, something completely unrelated to PX4, that publishes position setpoints in ROS frames using ROS standard messages. Regardless the frames mismatch, you still need to convert from standard ROS messages to PX4 messages, therefore an additional ROS node is needed, that node can also apply the rotations.

I would be happy to work on the docs for the different conventions (starting from https://docs.px4.io/main/en/tutorials/motion-capture.html#coordinate-frames)!

Reply:
> I would be happy to work on the docs for the different conventions (starting from https://docs.px4.io/main/en/tutorials/motion-capture.html#coordinate-frames)!

@beniaminopozzan That would be great. I think this information should be in the ROS2 User Guide because that is where  people will get a first view on conventions.
Possibly as a sub topic of [compatibility issues](https://docs.px4.io/main/en/ros/ros2_comm.html#compatibility-issues). Up to you.

I think having frames information would be good. Perhaps we should merge https://docs.px4.io/main/en/tutorials/motion-capture.html into https://docs.px4.io/main/en/ros/external_position_estimation.html then pull the https://docs.px4.io/main/en/ros/external_position_estimation.html#reference-frames-and-ros section into the ROS2 user guide and update it to be ROS2 rather than ROS1/MAVROS centric. Again, that's the direction I'd follow, but you're the expert.

Reply:
> I think this information should be in the ROS2 User Guide because that is where people will get a first view on conventions.

Yes, definitively!

> I think having frames information would be good. Perhaps we should merge https://docs.px4.io/main/en/tutorials/motion-capture.html into https://docs.px4.io/main/en/ros/external_position_estimation.html then pull the https://docs.px4.io/main/en/ros/external_position_estimation.html#reference-frames-and-ros section into the ROS2 user guide and update it to be ROS2 rather than ROS1/MAVROS centric. Again, that's the direction I'd follow, but you're the expert.

Yes, definitively x2!
And there is also https://docs.px4.io/main/en/advanced_config/tuning_the_ecl_ekf.html#external-vision-system :) !

Reply:
:-)

W.r.t. https://docs.px4.io/main/en/advanced_config/tuning_the_ecl_ekf.html#external-vision-system we should leave that there because it is the central place for all things EKF2. BUT we should also cross link to it from everywhere that needs information like this.

I'll keep an eye out for your PR on this! Do you need reminders?

Reply:
Of course!

> Do you need reminders?

No thanks! I just have to find the time to work on it :)